lighthouse-core/scripts/build-report-for-autodeployment.js

```diff
@@ -49,3 +48,68 @@ lhr.categories['lighthouse-plugin-someplugin'] = {
     }
   });
 })();
+
+/**
+ * Add a plugin to demo plugin rendering.
+ * @param {LH.Result} lhr
+ */
+function addPluginCategory(lhr) {
+  lhr.categories['lighthouse-plugin-someplugin'] = {
+    id: 'lighthouse-plugin-someplugin',
+    title: 'Plugin',
+    score: 0.5,
+    auditRefs: [],
+  };
+}
+
+/**
+ * Generate an LHR with errors for the renderer to display
+ * We'll write an "empty" artifacts file to disk, only to use it in auditMode
+ * @return {Promise<LH.Result>}
+ */
+async function generateErrorLHR() {
+  const url = 'http://fakeurl.com';
+  const options = {
+    requestedUrl: url,
+    settings: defaultSettings,
+    driver: {
+      getBrowserVersion: () => ({userAgent: 'Mozilla/5.0 ErrorUserAgent Chrome/66'}),
+    },
+  };
+  // @ts-ignore driver isn't mocked out completely
+  const artifacts = await GatherRunner.initializeBaseArtifacts(options);
```

Comment:
One downside here is that ts-ignore is going to make this invisible to any breaking changes made to initializeBaseArtifacts, and since it's an internal method that could pretty easily happen

Reply:
how about throwing a ts-ignore/cast on the driver object instead?